### PR TITLE
Remove dependency on Non-standard container images for instrumented tests

### PR DESCRIPTION
**Pull request by p3dr0rv**
## Why
From MS
Your service has been identified by Microsoft's Containers Secure Supply Chain (CSSC) initiative for using non-standard container images, which poses a threat to the security and reliability of your service and to Microsoft as a whole.
https://eng.ms/docs/more/containers-secure-supply-chain

## What change
Replace custom docker images and 1ES agents with Microsoft -hosted agents in order to be complaints.

### CODE REVIEW

File: azure-pipelines/auth-client/common-instrumented-tests.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# File: azure-pipelines\auth-client\common-instrumented-test.yml
```
Comment (iamgusain) [inline azure-pipelines/auth-client/common-instrumented-tests.yml:1]: Should we create these files in android-complete, to start consolidating our pipelines/yml files in one place.?

Comment (p3dr0rv) [reply azure-pipelines/auth-client/common-instrumented-tests.yml:1]: I f we want this pipeline to be triggered on every PR we need this file to be here. 
Trigger a pipeline from a resource only works for Azure repos :(
https://learn.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#define-a-repositories-resource

Comment (p3dr0rv) [reply azure-pipelines/auth-client/common-instrumented-tests.yml:1]: Tested here 
https://github.com/AzureAD/android-complete/pull/190
https://identitydivision.visualstudio.com/Engineering/_build?definitionId=1863

Comment (p3dr0rv) [reply azure-pipelines/auth-client/common-instrumented-tests.yml:1]: synced offline

File: azure-pipelines/auth-client/common-instrumented-tests.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+# Description: Run common instrumented test.
+
+trigger:
+- main
```
Comment (iamgusain) [inline azure-pipelines/auth-client/common-instrumented-tests.yml]: Should this be "dev"?

**Review by iamgusain — APPROVED** [review]

**Review by fadidurah — APPROVED** [review]

File: azure-pipelines/auth-client/common-instrumented-tests.yml (hunk 2 above)
Comment (p3dr0rv) [reply azure-pipelines/auth-client/common-instrumented-tests.yml]: done

**Pull request merged**